File: Atividades/Lista_2/Java/SimulacaoDeImpressora.java
package Atividades.Lista_2.Java;

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class SimulacaoDeImpressora {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        Impressora impressora = new Impressora();
        impressora.bluetooth = true;
        impressora.wifi = true;
        impressora.colorida = true;
        impressora.ligada = false;
        impressora.papel = true;
        impressora.digitalizadora = true;
        impressora.copiadora = false;

        impressora.modelo = input.next();
        impressora.cor = input.next();
        impressora.tipoPapel = input.next();

        impressora.status();

        input.close();
    }

}

class Impressora {
    public String modelo, cor, tipoPapel;
    public boolean bluetooth, wifi, colorida, ligada, papel, digitalizadora, copiadora;

    public void ligar() {
        if (!ligada) {
            System.out.println("ligando...");
            ligada = true;
        }
    }

    public void desligar() {
        if (ligada) {
            System.out.println("desligando...");
            ligada = false;
        }
    }

    public void imprimir() {
        if (ligada) {
            if (papel)
                System.out.println("imprimindo...");
            else
                System.out.println("sem papel");
        } else {
            System.out.println("impressora desligada");
        }
    }

    public void digitalizar() {
        if (ligada) {
            if (digitalizadora)
                System.out.println("digitalização sendo realizada...");
            else
                System.out.println("não é possível digitalizar");
        } else {
            System.out.println("impressora desligada");
        }
    }

    public void copiar() {
        if (ligada) {
            if (copiadora)
                System.out.println("cópia sendo realizada...");
            else
                System.out.println("não é possível copiar");
        } else {
            System.out.println("impressora desligada");
        }
    }

    public void status() {
        Map<Boolean, String> onOff = new HashMap<>();
        onOff.put(true, "on");
        onOff.put(false, "off");

        System.out.printf("%s\n%s\n%s\n",
            modelo, cor, tipoPapel
        );

        System.out.printf("bluetooth %s\nwifi %s\nimpressão %s\nimpressora %s\n%stem papel\ndigitalizadora %s\ncopiadora %s\n",
            onOff.get(bluetooth),
            onOff.get(wifi),
            colorida ? "colorida" : "preto e branco",
            onOff.get(ligada),
            papel ? "" : "não ",
            onOff.get(digitalizadora),
            onOff.get(copiadora)
        );
        
        ligar();
        digitalizar();
        copiar();
        desligar();
        imprimir();
    }

}

File: Atividades/Lista_2/C++/simulacaoDeImpressora.cpp
#include <iostream>
#include <string>
#include <unordered_map>

using namespace std;

class Impressora {
public:
    string modelo, cor, tipoPapel;
    bool bluetooth, wifi, colorida, ligada, papel, digitalizadora, copiadora;

    void ligar() {
        if (!ligada) {
            cout << "ligando..." << endl;
            ligada = true;
        }
    }

    void desligar() {
        if (ligada) {
            cout << "desligando..." << endl;
            ligada = false;
        }
    }

    void imprimir() {
        if (ligada) {
            if (papel)
                cout << "imprimindo..." << endl;
            else
                cout << "sem papel" << endl;
        } else {
            cout << "impressora desligada" << endl;
        }
    }

    void digitalizar() {
        if (ligada) {
            if (digitalizadora)
                cout << "digitalização sendo realizada..." << endl;
            else
                cout << "não é possível digitalizar" << endl;
        } else {
            cout << "impressora desligada" << endl;
        }
    }

    void copiar() {
        if (ligada) {
            if (copiadora)
                cout << "cópia sendo realizada..." << endl;
            else
                cout << "não é possível copiar" << endl;
        } else {
            cout << "impressora desligada" << endl;
        }
    }

    void status() {
        unordered_map<bool, string> onOff = {
            {true, "on"},
            {false, "off"}
        };

        cout << modelo << endl << cor << endl << tipoPapel << endl;

        cout << "bluetooth " << onOff[bluetooth] << endl;
        cout << "wifi " << onOff[wifi] << endl;
        cout << "impressão " << (colorida ? "colorida" : "preto e branco") << endl;
        cout << "impressora " << onOff[ligada] << endl;
        cout << (papel ? "" : "não ") << "tem papel" << endl;
        cout << "digitalizadora " << onOff[digitalizadora] << endl;
        cout << "copiadora " << onOff[copiadora] << endl;

        ligar();
        digitalizar();
        copiar();
        desligar();
        imprimir();
    }
};

int main() {
    Impressora impressora;

    impressora.bluetooth = true;
    impressora.wifi = true;
    impressora.colorida = true;
    impressora.ligada = false;
    impressora.papel = true;
    impressora.digitalizadora = true;
    impressora.copiadora = false;

    cin >> impressora.modelo;
    cin >> impressora.cor;
    cin >> impressora.tipoPapel;

    impressora.status();

    return 0;
}